### DISCUSSION TITLE
Using renovate and selectively changing what bits of the version string gets updated.

### DISCUSSION THREAD

**Discussion by ktasper** [renovatebot/renovate#30354] [Request Help] [answered]
### What would you like help with?

Using renovate and selectively changing what bits of the version string gets updated

### How are you running Renovate?

Locally, and in Gitlab . But this issue can be replicated locally.

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.
`37.440.6`

### Please tell us more about your question or problem

When attempting to use `renovate` for AWS Lambda Python 3 Images: https://gallery.ecr.aws/lambda/python

I have it set up like so:
```json
    {
      "groupName": "Custom versioning for lambda-python3* docker images",
      "matchPaths": [
        "docker/lambda-python311/Dockerfile"
      ],
      "extractVersion": "^3\\.\\d+\\.(?<version>.*)$",
      "versioning": "loose"
    }
```
With the value of `FROM public.ecr.aws/lambda/python:3.11.2024.01.01.01 as lambda-python311` 
However when I run renovate the new version comes back as `2024.07.10.11` (Its missing the python version at the beginning of the docker tag).

However If I remove the `extractVersion` we end up with a docker tag of `3.12.2024.07.10.11` so its bumped the python runtime as well as the `minor, major, patch`.

I want to be able to update `3.11.X.X.X` and not have to add in `3.11` each time renovate runs.

Any ideas what I can do to solve this?

**Comment by rarkins** [comment, marked as answer]
Try using versionCompatibility instead of extractVersion

**Reply by ktasper** [reply]
Yeah that seems to solve it.

```js
          matchPaths: [
            "docker/lambda-test/Dockerfile",
          ],
          versioning: "loose",
          versionCompatibility: "^(?<compatibility>3\\.\\d+\\.)(?<version>.*)$",

```
Thanks!
